packages/@aws-cdk/aws-route53-patterns/lib/website-redirect.ts

```diff
@@ -0,0 +1,73 @@
+import { DnsValidatedCertificate, ICertificate } from '@aws-cdk/aws-certificatemanager';
+import { CloudFrontWebDistribution, OriginProtocolPolicy, PriceClass, ViewerProtocolPolicy } from '@aws-cdk/aws-cloudfront';
+import { ARecord, IHostedZone, RecordTarget } from '@aws-cdk/aws-route53';
+import { CloudFrontTarget } from '@aws-cdk/aws-route53-targets';
+import { Bucket, RedirectProtocol } from '@aws-cdk/aws-s3';
+import { Construct, Fn, RemovalPolicy } from '@aws-cdk/core';
+
+export interface HTTPSRedirectProps {
+  /**
+   * HostedZone of the domain
+   */
+  readonly zone: IHostedZone;
+  /**
+   * The redirect target
+   */
+  readonly redirectTarget: string;
+  /**
+   * The domain name
+   *
+   * @default - the domain name of the zone
+   */
```

Comment:
Should this only be a subdomain? Will this also support redirecting the root domain?

Reply:
Perhaps call it recordName to align with route53 terminology? Provide an @example

Reply:
@hoegertn what do you think?